### PR TITLE
Building the front app, login and register part

### PR DESCRIPTION
**Pull request by Pepo995**
**- What I did**
Building the front app, login and register part

**- How I did it**

### CODE REVIEW

File: front-app/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules/
+dist/
+.prettierrc.cjs
+.eslintrc.cjs
+env.d.ts
```
Comment (MartinLarroca) [inline front-app/.eslintignore]: Extra line missing here

File: front-app/.eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      version: 'detect',
+    },
+    // Tells eslint how to resolve imports
+    'import/resolver': {
+      node: {
+        paths: ['src'],
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
+      },
+    },
+  },
```
Comment (MartinLarroca) [inline front-app/.eslintrc.cjs]: Let's remove comments

File: front-app/src/api/usersApi.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import axios from 'axios';
+import { RegisterValidationInterface } from '../views/Register';
+import { LoginValidationInterface } from '../views/Login';
+
+const usersApi = axios.create({
+  baseURL: 'http://localhost:4000/users',
```
Comment (MartinLarroca) [inline front-app/src/api/usersApi.tsx]: Let's move this to an environment variable

File: front-app/src/api/usersApi.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+export const getUsers = async () => {
+  const response = await usersApi.get('/');
+  return response.data;
+};
```
Comment (MartinLarroca) [inline front-app/src/api/usersApi.tsx]: Remember we almost always prefer this sin tax in this kind of situations

```suggestion
export const getUsers = async () => {
  const { data } = await usersApi.get('/');
  return data;
};
```

File: front-app/src/api/usersApi.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+export const createUser = (user: RegisterValidationInterface) => usersApi.post('/register', user);
+
+export const login = (user: LoginValidationInterface) => usersApi.post('/login', user);
```
Comment (MartinLarroca) [inline front-app/src/api/usersApi.tsx]: Semantically, for me it doesn't make sense that a variable called user is of type `LoginValidationInterface`. Two option here, or changing the interface name or the variable name. It's a personal opinion buy maybe renaming user to data, or loginData makes more sense. This goes for createUser and login too

File: front-app/src/components/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <p>${text}</p>
+  ) : (
+    <button
+      type='submit'
+      className='mt-6 h-9 sm:h-7 lg:h-9 w-full uppercase bg-linear-gradient-primary font-medium text-sm lg:text-base text-white bg-transparent rounded-md'
+    >
+      {children}
+    </button>
+  );
+};
```
Comment (MartinLarroca) [inline front-app/src/components/Button.tsx:20]: I don't really understand this component and how you are using it, let's discuss about it in the daily.

File: front-app/src/index.css [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  color: #0f58b7;
+}
+
+.bg-dark-blue {
+  background-color: #0f58b7;
+}
+
+.top-90 {
+  top: 90%;
+}
```
Comment (MartinLarroca) [inline front-app/src/index.css]: Are you using this class, don't think is necessary, you can do it by passing custom values to the top tailwind property.

File: front-app/src/layouts/LeftLayout.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import plan_it_logo from '../assets/svg/planit_logo.svg';
+import point from '../assets/svg/point.svg';
+import logo_p from '../assets/svg/logo_p.svg';
+import macbook from '../assets/img/macbook.png';
+import bg_blue from '../assets/img/image2.png';
+import logo_color from '../assets/svg/logo_and_letters.svg';
+import fb_logo from '../assets/svg/fb_gray.svg';
+import linkedin_logo from '../assets/svg/linkedin_gray.svg';
+import twitter_logo from '../assets/svg/twitter_gray.svg';
+import ig_logo from '../assets/svg/ig_gray.svg';
```
Comment (MartinLarroca) [inline front-app/src/layouts/LeftLayout.tsx]: It's okay if the file names are in snake case even though I think it's better to have them in PascalCase. But the names you are giving to the imports, those ones must be in PascalCase.

File: front-app/src/views/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { Link } from 'react-router-dom';
+import LeftLayout from '../layouts/LeftLayout';
+import TextInput from '../components/TextInput';
+
```
Comment (MartinLarroca) [inline front-app/src/views/Login.tsx:8]: Unnecessary space here

File: front-app/src/views/Login.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+          <h4 className='font-semibold sm:font-medium text-4xl sm:text-3xl lg:text-4xl'>¡Hola, otra vez!</h4>
+          <p className='text-lg sm:text-sm lg:text-lg font-medium sm:font-normal'>
+            Por favor, confirma tu información para ingresar a Plan IT.
+          </p>
+        </div>
+
+        <form
+          className='flex flex-col gap-6 lg:gap-8 '
+          onSubmit={handleSubmit((values) => {
+            console.log(values); //call the register endpoint
```
Comment (MartinLarroca) [inline front-app/src/views/Login.tsx]: Let's remove this console.log

File: front-app/src/views/Login.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    formState: { errors },
+  } = useForm<LoginValidationInterface>({
+    resolver: zodResolver(loginValidationSchema),
+  });
+
+  const { mutate: loginMutation, isLoading } = useMutation({
+    mutationFn: login,
+    onSuccess: (data) => {
+      console.log('User logged in!');
+      console.log(data.data.token);
```
Comment (MartinLarroca) [inline front-app/src/views/Login.tsx]: Remove console logs here

File: front-app/src/views/Login.tsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { Link } from 'react-router-dom';
+import LeftLayout from '../layouts/LeftLayout';
+import TextInput from '../components/TextInput';
+
+import { useMutation } from 'react-query';
+import { login } from '../api/usersApi';
+import { Button } from '../components/Button';
```
Comment (MartinLarroca) [inline front-app/src/views/Login.tsx:10]: Remember to create a folder for the Button component with the same conventions we are using

File: front-app/src/views/Register.tsx [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+        <div className='flex font-medium sm:font-normal text-sm lg:text-base text-gray-500'>
+          <p>¿Ya tenés una cuenta?</p>
+          <Link to='/login' className='text-dark-blue'>
+            Inicia sesión aquí.
+          </Link>
+        </div>
+        <form
+          className='flex flex-col gap-4 '
+          onSubmit={handleSubmit((values) => {
+            console.log(values);
```
Comment (MartinLarroca) [inline front-app/src/views/Register.tsx]: console.log here

**Review by MartinLarroca — COMMENTED** [review]
Great work pepo!!  Left you some comments. Also front-app/src/App.css file is empty, let's delete it

File: front-app/src/views/Login.tsx (hunk 12 above)
Comment (Pepo995) [reply front-app/src/views/Login.tsx:10]: Talked earlier, solved not to create folder

File: front-app/.env.example [hunk 14]
```diff
@@ -0,0 +1 @@
+PORT =
```
Comment (MartinLarroca) [inline front-app/.env.example]: Extra line missing here

File: front-app/src/api/usersApi.tsx [hunk 15]
```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require('dotenv').config();
```
Comment (MartinLarroca) [inline front-app/src/api/usersApi.tsx]: Don't think we need this, check the vite docs, it should be doing automatically

https://vitejs.dev/guide/env-and-mode.html

image

File: front-app/src/views/Login.tsx [hunk 16]
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { Link } from 'react-router-dom';
+import LeftLayout from '../layouts/LeftLayout';
+import TextInput from '../components/TextInput';
+import { useMutation } from 'react-query';
+import { login } from '../api/usersApi';
+import { Button } from '../components/Button';
```
Comment (MartinLarroca) [inline front-app/src/views/Login.tsx]: https://manurana.medium.com/organizing-imports-in-react-and-react-native-faf982a3a3b5

Check this, and try to config eslint to do it automatically

File: front-app/src/views/Register.tsx [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+            placeholder='Contraseña'
+            name='password'
+            hidden
+            error={!!errors.password}
+            errorMessage={errors.password?.message}
+          />
+          <TextInput register={register} placeholder='Nombre de su empresa' name='company' />
+          <Button loading={isLoading} text='Loading...'>
+            registrarse
+          </Button>
```
Comment (MartinLarroca) [inline front-app/src/views/Register.tsx]: Let's add some extra spacing here

File: front-app/.env.example (hunk 14 above)
Comment (MartinLarroca) [inline front-app/.env.example]: Still missing the extra line

Comment (Pepo995) [reply front-app/.env.example]: Screen Shot 2023-04-03 at 19 19 02

**Review by MartinLarroca — APPROVED** [review]

**Pull request merged**